queasars/job_shop_scheduling/problem_instances.py

```diff
@@ -0,0 +1,375 @@
+class ScheduledOperation:
+    """
+    Dataclass representing the fact that an Operation has been scheduled to start
+    at a certain time
+
+    :param operation: which has been scheduled
+    :type operation: Operation
+    :param start: time at which the operation has been scheduled to start. Can be None if the operation
+        was not scheduled
+    :type start: Optional[int]
+    """
+
+    operation: Operation
+    start: Optional[int]
+
+    @property
+    def is_scheduled(self) -> bool:
+        """
+        :return: whether the operation was successfully scheduled
+        :rtype: bool
+        """
+        return self.start is not None
+
+    @property
+    def start_time(self) -> int:
+        """
+        :return: the start time of the operation, according to the schedule
+        :rtype: int
+        :raise: JobShopSchedulingProblemException, if the operation was not successfully scheduled
+        """
+        if self.start is None:
+            raise JobShopSchedulingProblemException("Can't retrieve the start time if the operation was not scheduled!")
+        return self.start
+
+    @property
+    def end_time(self) -> int:
+        """
+        :return: the end time of the operation, according to the schedule
+        :rtype: int
+        :raise: JobShopSchedulingProblemException, if the operation was not successfully scheduled
+        """
+        if self.start is None:
+            raise JobShopSchedulingProblemException("Can't retrieve the end time if the operation was not scheduled!")
+        return self.start + self.operation.processing_duration
+
+    def __repr__(self):
+        if self.start is None:
+            return f"{str(self.operation)} was not or invalidly scheduled."
+        return f"{str(self.operation)} starts at: {self.start_time} and ends at: {self.end_time}"
+
+
+class JobShopSchedulingResult:
+    """
+    Class representing an attempted solution to the given job shop scheduling problem instance
+
+    :param problem_instance: for which this schedule represents an attempted solution
+    :type problem_instance: JobShopSchedulingProblemInstance
+    :param schedule: which attempts to solve the given problem instance
+    :type schedule: dict[Job, tuple[ScheduledOperation, ...]]
+    """
+
+    def __init__(
+        self, problem_instance: JobShopSchedulingProblemInstance, schedule: dict[Job, tuple[ScheduledOperation, ...]]
+    ):
+
+        if set(problem_instance.jobs) != set(schedule.keys()):
+            raise JobShopSchedulingProblemException(
+                "The JobShopSchedulingResult must contain the same Jobs "
+                + "as the problem instance which it is a solution to!"
+            )
+
+        for job in problem_instance.jobs:
+            if job.operations != tuple(map(lambda x: x.operation, schedule[job])):
+                raise JobShopSchedulingProblemException(
+                    "The schedule for a Job must contain the same operations as the Job itself!"
```

Comment:
Maybe add at least the name of the incorrect job here?

Reply:
Done :).